tests/conftest.py

```diff
@@ -204,3 +205,112 @@ def _R(expr):
         assert len(base) == 1
         base = base.pop()
     return EvalDerivative(*expr.args, base=base)
+
+
+# Utilities for testing tree structure
+
+
+def assert_structure(operator, exp_trees=None, exp_iters=None):
+    """
+    Utility function that helps to check loop structure of IETs. Retrieves trees from an
+    Operator and check that the blocking structure is as expected. Trees and Iterations
+    are returned for further use in tests.
+
+    Examples
+    --------
+    To check that an Iteration tree has the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x
+                for y
+            for f
+                for y
+
+    we call(Note: `time` mapped to `t`):
+
+    .. code-block:: python
+
+        trees, iters = assert_structure(op, ['t,x,y', 't,f,y'], 't,x,y,f,y')`
+    """
+    mapper = {'time': 't'}
+    trees = retrieve_iteration_tree(operator)
+    iters = FindNodes(Iteration).visit(operator)
+
+    if exp_trees is not None:
+        exp_trees = [i.replace(',', '') for i in exp_trees]  # 't,x,y' -> 'txy'
+        tree_struc = (["".join(mapper.get(i.dim.name, i.dim.name) for i in j)
+                       for j in trees])
+        assert tree_struc == exp_trees
+
+    if exp_iters is not None:
+        exp_iters = exp_iters.replace(',', '')  # 't,x,y' -> 'txy'
+        iter_struc = "".join(mapper.get(i.dim.name, i.dim.name) for i in iters)
+        assert iter_struc == exp_iters
+
+    return trees, iters
+
+
+def assert_blocking(operator, exp_nests):
+    """
+    Utility function that helps to check existence of blocked nests. The structure of the
+    operator is not used.
+
+    Examples
+    --------
+    For the following structure:
+
+    .. code-block:: python
+
+        for time
+            for x0_blk0
+                for x
+            for x1_blk0
+                for x
+
+    we call:
+
+    .. code-block:: python
+
+       bns, pbs = assert_blocking(op, {'x0_blk0', 'x1_blk0'})
+
+    to assert the existence of 'x0_blk0', 'x1_blk0' and then the function returns a
+    dictionary with the blocking Iterations that start a blocking subtree:
+
+    ['x0_blk0': Iteration x0_blk0..., 'x1_blk0': Iteration x1_blk0...]
+
+    and the ParallelBlock that encapsules the above blocking subtree
+
+    ['x0_blk0': ParallelBlock encapsulating Iteration x0_blk0,
+     'x1_blk0': ParallelBlock encapsulating Iteration x1_blk0]
+    """
+    bns = {}
+    pbs = {}
+    trees = retrieve_iteration_tree(operator)
+    for tree in trees:
+        iterations = [i for i in tree if i.dim.is_Incr]  # Collect Incr dimensions
+        parallel_blocks = FindNodes(ParallelBlock).visit(tree)
```

Comment:
You only need this tree traversal if you get inside the `if iterations` branch right? so move it below inside the `try`?